### PR TITLE
drivers: sensor: stm32_temp: Add multi-instance support

### PR DESCRIPTION
**Pull request by benjaminbjornsson**
Move driver to use DT_INST_FOREACH_STATUS_OKAY to add
multi-instance support.

Signed-off-by: Benjamin Björnsson <benjamin.bjornsson@gmail.com>

### CODE REVIEW

**Review by gmarull — COMMENTED** [review]
Do STM32 have > 1 instance?

**Comment by benjaminbjornsson** [comment]
> Do STM32 have > 1 instance?

My thought was that multi-instance support doesn’t hurt and that it might be useful in other ways than for running on real hardware.

File: drivers/sensor/stm32_temp/stm32_temp.c [hunk 1]
```diff
@@ -6,6 +6,7 @@
  */
 
 #include <zephyr/device.h>
+#include <devicetree.h>
```
Comment (gmarull) [inline drivers/sensor/stm32_temp/stm32_temp.c:9]:
```suggestion
#include <zephyr/devicetree.h>
```

Comment (benjaminbjornsson) [reply drivers/sensor/stm32_temp/stm32_temp.c]: Fixed!

**Review by gmarull — APPROVED** [review]

**Review by MaureenHelm — APPROVED** [review]

**Pull request merged**